File: examples/plugin-archives/demo-conditional-ui/python/provider.py
def execute(params, upstream, context):
    value = float(upstream or 0)
    if params.get("mode") == "shift":
        preset = params.get("preset")
        offset = 1 if preset == "plus1" else 5 if preset == "plus5" else float(params.get("offset", 0))
        return value + offset
    preset = params.get("preset")
    factor = 2 if preset == "double" else 3 if preset == "triple" else float(params.get("factor", 1))
    return value * factor

File: examples/plugin-archives/demo-conditional-ui/js/provider.js
function execute(params, upstream) {
  const value = Number(upstream ?? 0);
  if (params.mode === "shift") {
    const offset = params.preset === "plus1" ? 1 : params.preset === "plus5" ? 5 : Number(params.offset ?? 0);
    return value + offset;
  }
  const factor = params.preset === "double" ? 2 : params.preset === "triple" ? 3 : Number(params.factor ?? 1);
  return value * factor;
}
